Tasks/Common/nuget-task-common/NuGetToolRunner2.ts

```diff
@@ -149,8 +178,25 @@ export async function getNuGetQuirksAsync(nuGetExePath: string): Promise<NuGetQu
 // identity's token.
 // Therefore, we are enabling credential provider on on-premises and disabling it on hosted (only when the version of NuGet does not support it). We allow for test
 // instances by an override variable.
+export function isCredentialProviderV1Enabled(quirks: NuGetQuirks): boolean {
+    if (quirks.hasQuirk(NuGetQuirkName.V2CredentialProvider)) {
```

Comment:
Checking the quirks first overrides all the other checks done below, especially the build var that would let us set it to true or false. I would just add this check to the end of the other checks below.